src/main/java/com/google/codeu/servlets/RestaurantDataServlet.java

```diff
@@ -43,7 +47,40 @@ public void init() {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("application/json");
-    response.getOutputStream().println(restaurantArray.toString());
+    
+    double parameterLatitude = Double.parseDouble(request.getParameter("latitude"));
+    double parameterLongitude = Double.parseDouble(request.getParameter("longitude"));
+    JsonArray filteredRestaurantArray = filter(parameterLatitude, parameterLongitude);
+      
+    response.getOutputStream().println(filteredRestaurantArray.toString());
+  }
+    
+  private JsonArray filter(double parameterLatitude, double parameterLongitude) {
+    JsonArray filteredArray = new JsonArray();
+    for (JsonElement restaurant : restaurantArray) {
+      JsonObject restaurantObject = restaurant.getAsJsonObject();
+      double latitude = restaurantObject.get("lat").getAsDouble();
+      double longitude = restaurantObject.get("lng").getAsDouble();
+      if (distance(parameterLatitude, parameterLongitude, latitude, longitude) <= defaultRadius) {
+        //System.out.println("in range");
+        filteredArray.add(restaurant);
+      }
+    }
+    return filteredArray;
+  }
+    
+  private double distance(double lat1, double lon1, double lat2, double lon2) {
+    if ((lat1 == lat2) && (lon1 == lon2)) {
+      return 0;
+    }
+    else {
+      double theta = lon1 - lon2;
```

Comment:
Thanks for researching this! I was just thinking of using the Pythagorean theorem, but upon my own searches online, I see that this formula is more accurate.